### PR TITLE
Fixed #2035 issue to ensure user gets an notification when he clicks the 'SAVE CHANGES' button present in the settings tab.

### LINKED ISSUE #2035: Notification for User for the Saved Changes in Settings Tab
**Actual Behaviour**
No notification for the user if his changes are saved or not when clicked on the 'SAVE CHANGES' button in Settings tab..
Demo can be viewed at:-
https://s23.aconvert.com/convert/p3r68-cdx67/ik6kb-hhryq.gif

**Expected Behaviour**
There should be a clarification for the user if his changes are saved when the button is clicked..This ensures a better usability of the model.
Demo can be seen at:-
https://s17.aconvert.com/convert/p3r68-cdx67/ktyjb-xez32.gif

**Steps to reproduce it**
Go to the Settings Tab from the drop down list and see the result on clicking 'SAVE CHANGES' button from any of the present entities like Account, Password, Mobile, etc..No effects on clicking the 
'SAVE BUTTON' making it tedious for the user to understand if his/her changes are manipulated or not.. 

**Would you like to work on the issue?**
Yes..

### PR DESCRIPTION
**Pull request by rajshah9914**
Fixes #2035 

Changes: 
A notification for the user is added such that he/she gets the clarification when SAVE CHANGES' button in the settings drop down is clicked.
Actual working:
Demo can be viewed at:-
original

This is a bit user friendly:-
Demo can be seen at:-
expected

### CODE REVIEW

**Comment by rajshah9914** [comment]
@plxity I have updated the changes and followed the PR template as you mentioned earlier.

**Comment by rajshah9914** [comment]
Ok @AakashMallik will replace it soon..

**Comment by rajshah9914** [comment]
snackbar_notification
@AakashMallik , I have added the requested SnackBar instead of a JS Alert..

File: src/components/ChatApp/Settings/Settings.react.js [hunk 1]
```diff
@@ -626,6 +626,11 @@ class Settings extends Component {
 
   // Submit selected Settings
   handleSubmit = () => {
+    var x = document.getElementById('snackbar');
```
Comment (AakashMallik) [inline src/components/ChatApp/Settings/Settings.react.js]: Please don't make use of vanilla JS code. We have a well-placed view lib just for this purpose. Refer to other files where snackbar is being used and try to use a similar code pattern.

Comment (rajshah9914) [reply src/components/ChatApp/Settings/Settings.react.js]: Ok, would make the changes soon..

**Comment by rajshah9914** [comment]
@AakashMallik Added snackbar with routes..
with_snackbar

**Comment by rajshah9914** [comment]
ScreenShot of Snackbar notification on the deployed surge link. 
deployed_site
Surge Link for Demo:-
http://chatbot_snackbar.surge.sh/

**Comment by rajshah9914** [comment]
> Not sure if we want snackbar, looks unnecessary

@fragm3 snackbar though may not be required, but a notification for the user should be there for changes that are getting saved according to me.. Now since many other notifications are through a snackbar , so is case here..And it actually looks better than a JS alert.

**Comment by AakashMallik** [comment]
@rajshah9914 I guess what he meant was the whole PR is not required.

**Comment by rajshah9914** [comment]
I myself wasn't sure if my changes get saved while changing tabs in Settings..so I guess this is actually needed..@AakashMallik

**Comment by AakashMallik** [comment]
@rajshah9914 I am in support for the change... makes sense to me to give a user a feedback.

**Comment by AakashMallik** [comment]
@rajshah9914 But I have an interesting idea. When user clicks save, use a spinner in the save button like we have in login. IF save fails, the show a snackbar.

**Comment by rajshah9914** [comment]
Ok will try that too and send the result..

**Comment by AakashMallik** [comment]
@rajshah9914 Cool

**Comment by rajshah9914** [comment]
> @rajshah9914 But I have an interesting idea. When user clicks save, use a spinner in the save button like we have in login. IF save fails, the show a snackbar.

@AakashMallik , I have a doubt, as you said to display a snackbar when saving fails, will do that but I having issues for saving and the spinner. When the user clicks 'Save Changes', there is no much time difference between Snackbar and the spinner motion, thus it happens rapidly. Actually on Login page, we are directing user to home page, thus spinner motion suits there and an snackbar is displayed after login.. Do you think a spinner implementation is needed? Would try something better then or move to some other issue if a simple feedback through snackbar will suffice(demo in above comments)..

**Comment by rajshah9914** [comment]
I am trying out since our talk but results are not that great..Actually in login, we were having conditions for valid email , passwords, but here just toggle button, nothing to compare, so getting a bit cumbersome..not getting proper implementation for this, ie. when should I stop the spinner and display snackbar and so ..

**Comment by AakashMallik** [comment]
> I am trying out since our talk but results are not that great..Actually in login, we were having conditions for valid email , passwords, but here just toggle button, nothing to compare, so getting a bit cumbersome..not getting proper implementation for this, ie. when should I stop the spinner and display snackbar and so ..

Ask me the right questions and I will help you out.. :wink:

**Comment by rajshah9914** [comment]
> @rajshah9914 Please go ahead and display the feedback on a snackbar. Show suitable messages as per the API response.

Ok @akshatnitd Sir..will do it soon..

**Comment by rajshah9914** [comment]
Sure @akshatnitd Sir, left it due to my midsems, would do it soon..

**Comment by akshatnitd** [comment]
@rajshah9914 I hope your midsems must be over by now. Please finalise the PR.

**Comment by rajshah9914** [comment]
Oh, really sorry @akshatnitd  Sir, will do it by tomorrow.. most probably today itself.. Once again I apologize for delays..

**Comment by rajshah9914** [comment]
microphone
speech
theme


Added appropriate snackbars where ever necessity persisted.. @akshatnitd  Please review it..

**Comment by harsh-jindal** [comment]
@akshatnitd @rajshah9914 I don't think you need a snackbar message on toggle input. Like the user knows visually that he/she has turned on or off the setting from the toggle itself.
But if others think that it is required then you need to add this to ChatApp(Send message by pressing ENTER) too.

**Comment by rajshah9914** [comment]
@harsh-jindal , The situation is such that a snackbar cannot display all changes done on the page together when a 'Save changes' button is clicked. So, I rather displayed each and every continuous changes that a user performs as per what @akshatnitd had told.(To display feedback on snackbar for each change made.)

**Comment by fragm3** [comment]
Closing, changes implemented in https://github.com/fossasia/chat.susi.ai/pull/2129

**Pull request closed without merging**